test/units/plugins/strategy/test_strategy.py

```diff
@@ -470,6 +479,8 @@ def _queue_put(item, *args, **kwargs):
 
         mock_inc_file._filename = "test.yml"
         res = strategy_base._load_included_file(included_file=mock_inc_file, iterator=mock_iterator)
+        self.assertEqual(len(res), 1)
+        self.assertTrue(isinstance(res[0], Block))
```

Comment:
The code change actually revealed an issue with this unit test.

The mocks caused the parent task to have `collections` set as a `MagicMock` instance resulting in `debug` not being found and `_load_included_file` throwing `AnsibleParserError` which was then shadowed by `AnsibleError` and `_load_included_file` returned an empty list. The `res` was not asserted in any way and the test passed.

With this PR `AnsibleParserError` is thrown instead of being shadowed by `AnsibleError`, the unit test failed with `AnsibleParserError`. Now the mocks are "fixed" and the result checked for expected result.